Tools/Scripts/layout-test-failure-analysis

```diff
@@ -0,0 +1,291 @@
+# 1.  Redistributions of source code must retain the above copyright
+#     notice, this list of conditions and the following disclaimer.
+# 2.  Redistributions in binary form must reproduce the above copyright
+#     notice, this list of conditions and the following disclaimer in the
+#     documentation and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS BE LIABLE FOR ANY
+# DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+# (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+# ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+
+import argparse
+import logging
+import os
+import sys
+
+import re
+import requests
+from urllib.parse import urlparse
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(description='Export test names in the order they were run for a specific worker from stdio input.')
+
+    # data source
+    input_group = parser.add_mutually_exclusive_group(required=True)
+    input_group.add_argument('-f', '--file',
+                        dest='input_file',
+                        help='Specifies the file to read for layout test results data.',
+                        type=str)
+    input_group.add_argument('-u', '-url',
+                        dest='input_url',
+                        help='Specifies the URL from which to retrieve layout test results data.',
+                        type=str)
+
+    # optional flags
+    parser.add_argument('-t', '--test-path',
+                        dest='test_path',
+                        help='Name of the test for which to search. Omitting the test name will export all workers as separate files (unless --combine-workers is used).',
+                        default='',
+                        type=str)
+    parser.add_argument('-o', '--output-path',
+                        dest='output_path',
+                        help='Specifies the file name for the tests output file. Defaults to run_tests.txt in current directory. If there will be multiple output files (workers), this will be treated as a directory.',
+                        default='run_tests.txt',
+                        type=str)
+    parser.add_argument('-l', '--long',
+                        action='store_true',
+                        help='Lengthens the worker output such that it doesn\'t truncate at the requested test. Good for getting a full test list. This has no effect when the test name is omitted.',
```

Comment:
I'm a bit confused why this is an option. Are we trying to save network bandwidth by not downloading the entire file?

Reply:
The idea is that we can export the worker in full instead of truncating at a test if one isn't specified.